x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/plugin/SqlMediaTypeParserTests.java

```diff
@@ -104,6 +106,26 @@ public void testNoFormat() {
         assertEquals(e.getMessage(), "Invalid request content type: Accept=[null], Content-Type=[null], format=[null]");
     }
 
+    public void testCborClearCursor() {
+        assertEquals(
+            XContentType.JSON,
+            getResponseMediaType(reqWithAccept("application/json"), createClearCursorInstance(false, Mode.JDBC))
+        );
+        assertEquals(
+            XContentType.CBOR,
+            getResponseMediaType(reqWithAccept("application/json"), createClearCursorInstance(true, Mode.JDBC))
+        );
+        assertEquals(
+            XContentType.CBOR,
+            getResponseMediaType(reqWithAccept("application/whatever"), createClearCursorInstance(true, Mode.JDBC))
+        );
+        assertEquals(
+            XContentType.CBOR,
+            getResponseMediaType(reqWithAccept("application/json"), createClearCursorInstance(true, Mode.ODBC))
+        );
+        assertEquals(XContentType.CBOR, getResponseMediaType(reqWithAccept("application/json"), createClearCursorInstance(true, Mode.CLI)));
```

Comment:
Might be worth adding another test showing that for non-drivers the CBOR is not enforced, when no binary format is provided.

Reply:
Done